### PR TITLE
feat: add  "Add filters" to create alert screen

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-368] [ONYX-371] 

### Description

This PR makes the following changes:
- [x] Add feature flag
- [x] Add "Add filters" button to Create an alert screen.
- [x] Update the UI to match the new designs (I created a follow-up ticket)
- [ ] Navigate to "Create Alert" instead of the "Filter & Sort" modal on the Create Alert press 

Group 1 (7)

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add "Add Filters" to Create Alert modal - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-368]: https://artsyproduct.atlassian.net/browse/ONYX-368?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ONYX-371]: https://artsyproduct.atlassian.net/browse/ONYX-371?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ONYX-369]: https://artsyproduct.atlassian.net/browse/ONYX-369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Add "Add Filters" to Create Alert modal - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c13412813c294057749c651ed456161d535b3e55
</p>

**Review by nickskalkin — APPROVED** [review]
👍

File: src/app/Scenes/SavedSearchAlert/Components/Form.tsx [hunk 1]
```diff
@@ -132,91 +136,119 @@ export const Form: React.FC<FormProps> = ({
...
-              onPress={() => onRemovePill(pill)}
-            >
-              {pill.label}
-            </Pill>
-          ))}
-        </Flex>
-      </Box>
-      {!!priceControlEnabled && (
-        <>
+      <Join separator={enableAlertsFilters ? <Separator my={1} /> : <></>}>
```
Comment (dblandin) [inline src/app/Scenes/SavedSearchAlert/Components/Form.tsx:139]: **suggestion / blocking**: Are the separator lines intentional? I'm not seeing those in [the design spec](https://www.figma.com/file/lVbIxbnH2IUk1K1FoxyFHV/Filters-and-Alerts-redesign?type=design&node-id=250-31044&mode=design&t=KdFXpvkDX4ClHtNm-4):
Screenshot 2023-09-28 at 14 29 02

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/Form.tsx:139]: 🤯

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/Form.tsx:139]: took me a while to get them to show up properly 😄 i am sure there were separators but can't find them anymore 😕

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/Form.tsx:139]: I will remove them because I can't find them anymore

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/Form.tsx:139]: I was looking at mobile web 😢

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/Form.tsx]: done ✅

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/Components/Form.tsx]: Uploading Screen Recording 2023-09-28 at 11.21.11.mov…

**Pull request merged**